### DISCUSSION TITLE
How to log requests from rewrites in console?

### DISCUSSION THREAD

**Discussion by ivoiv** [vercel/next.js#45710] [Help] [answered]
I want want to see what/how nextjs does to the request object during rewrites.
How do I log those to the terminal/console?

**Comment by gustavostz** [comment, marked as answer]
You can use the express.js and app.use middleware in the pages/_app.js file. Like this example:

```javascript
import App from 'next/app'
import express from 'express'

const app = express()

app.use((req, res, next) => {
  console.log('Request object during rewrite: ', req.url)
  next()
})

export default class MyApp extends App {
  ...
}
```

As you can see, in this example, the express.js middleware logs the URL of the request to the console whenever a rewrite occurs in Next.js. The next() function is called to continue processing the request and sending the response

**Reply by maciek-mckinsey** [reply]
Is there a way to achieve something similar with app router?
